### PR TITLE
[Qt] Update to 6.6.0

### LINKED ISSUE #34389: [Qt] update to 6.6.0
### Library name

Qt

### New version number

6.6.0

### Other information that may be useful (release notes, etc...)

https://code.qt.io/cgit/qt/qtreleasenotes.git/about/qt/6.6.0/release-note.md

https://github.com/qt/qtbase/releases/tag/v6.6.0

https://wiki.qt.io/Qt_6.6_Release

### PR DESCRIPTION
**Pull request by Neumann-A**
closes  #34389

### CODE REVIEW

**Comment by Neumann-A** [comment]
@autoantwort if you are bored you can check if I need to add some `CMAKE_DISABLE_FIND_PACKAGES_ ` vars somewhere. At least libresolv needs to be added as it seems. (I already know how to fix qtgrpc but need a bit of time to prepare the patches and complain upstrream)

File: ports/qtbase/fix-host-aliasing.patch [hunk 1]
```diff
@@ -0,0 +1,17 @@
+diff --git a/cmake/QtTargetHelpers.cmake b/cmake/QtTargetHelpers.cmake
+index 180ec33..a03e3b1 100644
+--- a/cmake/QtTargetHelpers.cmake
++++ b/cmake/QtTargetHelpers.cmake
+@@ -281,10 +281,10 @@ function(qt_internal_add_target_aliases target)
+     set_target_properties("${target}" PROPERTIES _qt_versionfull_alias "${versionfull_alias}")
+ 
+     get_target_property(type "${target}" TYPE)
+-    if (type STREQUAL EXECUTABLE)
++    if (type STREQUAL EXECUTABLE AND NOT QT_HOST_PATH AND NOT TARGET "${versionfull_alias}")
```
Comment (Neumann-A) [inline ports/qtbase/fix-host-aliasing.patch:10]:
```suggestion
+    if (type STREQUAL EXECUTABLE AND NOT TARGET "${versionfull_alias}")
```

**Comment by carsten-grimm-at-ipolog** [comment]
Thanks for fighting your way through this, @Neumann-A !

**Comment by MonicaLiu0311** [comment]
When I tested feature `vulkan`, the following error occurred:
```
PS G:\qt> ./vcpkg install qtbase[vulkan]:x64-windows

...

-- Configuring x64-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: "C:/Program Files/Microsoft Visual Studio/2022/Enterprise/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" -v
    Working Directory: G:/qt/buildtrees/qtbase/x64-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      G:\qt\buildtrees\qtbase\config-x64-windows-dbg-CMakeCache.txt.log
      G:\qt\buildtrees\qtbase\config-x64-windows-rel-CMakeCache.txt.log
      G:\qt\buildtrees\qtbase\config-x64-windows-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:252 (vcpkg_execute_required_process)
  ports/qtbase/cmake/qt_install_submodule.cmake:147 (vcpkg_cmake_configure)
  ports/qtbase/cmake/qt_install_submodule.cmake:341 (qt_cmake_configure)
  ports/qtbase/portfile.cmake:292 (qt_install_submodule)
  scripts/ports.cmake:168 (include)
```

**G:\qt\buildtrees\qtbase\config-x64-windows-out.log:**
```
CMake Error at C:/Program Files/Microsoft Visual Studio/2022/Enterprise/VC/vcpkg/scripts/buildsystems/vcpkg.cmake:853 (_find_package):
  Could not find a package configuration file provided by "WrapVulkanHeaders"
  with any of the following names:

    WrapVulkanHeadersConfig.cmake
    wrapvulkanheaders-config.cmake

  Add the installation prefix of "WrapVulkanHeaders" to CMAKE_PREFIX_PATH or
  set "WrapVulkanHeaders_DIR" to a directory containing one of the above
  files.  If "WrapVulkanHeaders" provides a separate development package or
  SDK, be sure it has been installed.
Call Stack (most recent call first):
  cmake/QtFindPackageHelpers.cmake:107 (find_package)
  src/gui/configure.cmake:61 (qt_find_package)
  cmake/QtFeature.cmake:674 (include)
  src/CMakeLists.txt:13 (qt_feature_evaluate_features)


-- Configuring incomplete, errors occurred!
```

**Comment by Neumann-A** [comment]
> When I tested feature vulkan, the following error occurred:

You need to have vulkan installed. It is a system dependency. I added a dep on the stub `vulkan` port for it. The feature probably existed before the vulkan port existed.

**Comment by MonicaLiu0311** [comment]
> > When I tested feature vulkan, the following error occurred:
> 
> You need to have vulkan installed. It is a system dependency. I added a dep on the stub `vulkan` port for it. The feature probably existed before the vulkan port existed.

I did install vulkan, this problem only occurs on the current PR because I successfully tested the existing `qtbase[vulkan]` in vcpkg.

**Comment by Neumann-A** [comment]
> I did install vulkan, this problem only occurs on the current PR because I successfully tested the existing qtbase[vulkan] in vcpkg.

should be fixed now. I forgot that Qt is searching in CONFIG mode first so no MODULE packages can ever be made REQUIRED.

**Review by MonicaLiu0311 — APPROVED** [review]

**Pull request merged**